### ISSUE TITLE
Add Nodatime support

### ISSUE THREAD

**Issue by Hawxy** [rebus-org/Rebus.PostgreSql#12]
Nodatime is the [recommended way](https://www.npgsql.org/doc/types/nodatime.html) to handle date/time type mappings whilst using npgsql, so it's pretty common for people to use it.

However, due to a [long-standing issue](https://github.com/npgsql/npgsql/issues/2439), you cannot combine BCL types and nodatime together in the same application, so if nodatime support is enabled and you publish a message whilst using this package, it throws a `System.InvalidCastException: Can't write CLR type System.TimeSpan with handler type IntervalHandler`

In my scenario I'm using [Marten](https://github.com/JasperFx/marten) with Nodatime support enabled and am running into this problem.

**Comment by mookid8000** [comment]
When (and how?) does this exception happen when using Rebus?

**Comment by Hawxy** [comment]
Here's a minimal repo using:
- .NET Core 3.0
- Npgsql & Npgsql.NodaTime v4.1.1
- Rebus v5.4.0
- Rebus.PostgreSql 6.0.0-b01

```
    class Program
    {
        static async Task Main(string[] args)
        {
            var pgConn = "<connstr>";
            //Either intentionally called by the user or by a third-party postgres plugin
            NpgsqlConnection.GlobalTypeMapper.UseNodaTime();

            using var activator = new BuiltinHandlerActivator();
            activator.Handle<Something>(async x => Console.WriteLine("Something handled"));
            Configure.With(activator)
                .Transport(t => t.UsePostgreSql(new PostgresConnectionHelper(pgConn), "messages", "queue"))
                .Start();

            await activator.Bus.SendLocal(new Something());

            Console.WriteLine("Press enter to quit");
            Console.ReadLine();
        }
    }

    public class Something { }
```

From what I can see, the exception is thrown in the transport's `InnerSend` due the `TimeSpan` being set.

As a side note, I tried testing with Rebus v6.0.0-b18 but it's currently incompatible due to RebusTime changes.

**Comment by mookid8000** [comment]
hmmm, ok .... so, to support Noda, Rebus.Postgresql would have to

a) reference the nodatime NuGet package
b) be able to run in either of two modes: 1 BCL mode, 2 Noda mode
c1) be able to detected whether Npgsql is using Noda, or
c2) be configurable from the outside – something like `t => t.UsePostgreSql(..., useNodaTime: true)`

If that sounds right, then I guess that is a fairly straightforward task....

**Comment by Hawxy** [comment]
> a) reference the nodatime NuGet package

If you want to avoid this, then you might be interested in how Marten did it.

They have a [static class](https://github.com/JasperFx/marten/blob/0839ec6d364cdfd07ead7ccba15785ae29e72cb8/src/Marten/Util/TypeMappings.cs#L19) that allows for custom type mappings to be added, and then [extension methods](https://github.com/JasperFx/marten/blob/0839ec6d364cdfd07ead7ccba15785ae29e72cb8/src/Marten/Util/TypeMappings.cs#L344) that are called if they're [saving/reading a type](https://github.com/JasperFx/marten/blob/0ccc54748849236c49ccb38fa3f1984b4a2f6efe/src/Marten/Events/EventQueryHandler.cs#L66) that would need to be mapped to the compatible type when nodatime is enabled.

In this scenario you could publish a `Rebus.PostgreSql.NodaTime` package that simply sets the correct converters, [like this](https://github.com/JasperFx/marten/blob/0839ec6d364cdfd07ead7ccba15785ae29e72cb8/src/Marten.NodaTime/NodaTimeExtensions.cs#L22).

**Comment by Hawxy** [comment]
This has been fixed as of npgsql v4.1.5
